docs/sources/flow/reference/components/loki.source.heroku.md

```diff
@@ -0,0 +1,136 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/loki.source.heroku
+title: loki.source.heroku
+---
+
+# loki.source.heroku
+
+`loki.source.heroku` listens for Heroku messages over TCP connections
+and forwards them to other `loki.*` components.
+
+The component starts a new heroku listener for the given `listener`
+block and fans out incoming entries to the list of receivers in `forward_to`.
+
+Before using `loki.source.heroku`, Heroku should be configured with the URL where the Agent will be listening. Follow the steps in [Heroku HTTPS Drain docs](https://devcenter.heroku.com/articles/log-drains#https-drains) for using the Heroku CLI with a command like the following:
+
+```
+heroku drains:add [http|https]://HOSTNAME:PORT/heroku/api/v1/drain -a HEROKU_APP_NAME
+```
+
+Multiple `loki.source.heroku` components can be specified by giving them
+different labels.
+
+## Usage
+
+```river
+loki.source.heroku "LABEL" {
+  listener {
+    address = "LISTEN_ADDRESS"
+    port = PORT
+  }
+  ...
```

Comment:
We can't have more than one `listener` block, so can we remove the ellipses?

Reply:
Also, similar to other components, do we want this as a block, or have we considered flattening the structure?